spec/ameba/rule/lint/useless_assign_spec.cr

```diff
@@ -475,6 +475,32 @@ module Ameba::Rule::Lint
         issue.location.to_s.should eq "source.cr:1:1"
         issue.message.should eq "Useless assignment to variable `foo`"
       end
+
+      it "doesn't report if top level variable defined inside class is referenced" do
+        s = Source.new %(
+          class A
+            foo : String? = "foo"
+          end
+
+          puts foo
```

Comment:
Shouldn't this be put _inside_ the class declaration?

Reply:
Nope, we try to match the original scenario in the opened issue